src/core/core.tooltip.js

```diff
@@ -963,13 +963,24 @@ class Tooltip extends Element {
 	handleEvent(e) {
 		const me = this;
 		const options = me.options;
+		const mode = options.mode;
 		const lastActive = me._active || [];
 		let changed = false;
 		let active = [];
 
 		// Find Active Elements for tooltips
 		if (e.type !== 'mouseout') {
-			active = me._chart.getElementsAtEventForMode(e, options.mode, options);
+			active = me._chart.getElementsAtEventForMode(e, mode, options);
+
+			if (mode === 'dataset' && active.length) {
+				const datasetIndex = active[0].datasetIndex;
+				const meta = me._chart.getDatasetMeta(datasetIndex);
+				active = [];
```

Comment:
should we just return this from `getElementsAtEventForMode` instead?

Reply:
Yeah, feels like should have returned the correct thing rather than fixing it here